util/reggen/countermeasure.py

```diff
@@ -0,0 +1,69 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, List
+
+from .lib import check_keys, check_str, check_list
+
+# TO BE DEFINED
+CM_ASSETS = [
+    'FOO'
+]
+
+# TO BE DEFINED
+CM_TYPES = [
+    'BAR'
+]
+
+
+class CounterMeasure:
+    def __init__(self, asset: str, cmtype: str, desc: str):
+        self.asset = asset
+        self.cmtype = cmtype
+        self.desc = desc
+
+    @staticmethod
+    def from_raw(what: str, raw: object) -> 'CounterMeasure':
+        rd = check_keys(raw, what, ['name', 'desc'], [])
+
+        name = check_str(rd['name'], 'name field of ' + what)
+        desc = check_str(rd['desc'], 'desc field of ' + what)
+
+        try:
+            asset, cmtype = name.split('.')
+        except ValueError:
+            raise ValueError('The countermeasure ID {} ({}) has the '
+                             'wrong format.'
+                             .format(name, what))
+
+        if asset not in CM_ASSETS:
+            raise ValueError('The asset identifier {} of countermeasure '
+                             '{} ({}) is not defined.'
+                             .format(asset, name, what))
+
+        if cmtype not in CM_TYPES:
+            raise ValueError('The type identifier {} of countermeasure '
+                             '{} ({}) is not defined.'
+                             .format(cmtype, name, what))
+        return CounterMeasure(asset, cmtype, desc)
+
+    @staticmethod
+    def from_raw_list(what: str,
+                      raw: object) -> List['CounterMeasure']:
+        ret = []
+        for idx, entry in enumerate(check_list(raw, what)):
+            entry_what = 'entry {} of {}'.format(idx, what)
+            cm = CounterMeasure.from_raw(entry_what, entry)
+            ret.append(cm)
+        return ret
+
+    def _asdict(self) -> Dict[str, object]:
+        return {
+            'asset': self.asset,
+            'cmtype': self.cmtype,
```

Comment:
If you want this to round-trip, you'll want to glue the name back together again here.

Reply:
Ack - fixed.